BizHawk.Emulation.DiscSystem/DiscTypes.cs

```diff
@@ -105,29 +105,26 @@ public static bool IsMatch(string str)
 
 		/// <summary>
 		/// creates a timestamp from a string in the form mm:ss:ff
+		/// TODO use a datetime library
 		/// </summary>
 		public Timestamp(string str)
 		{
-			if (str.Length != 8) goto BOGUS;
-			if (str[0] < '0' || str[0] > '9') goto BOGUS;
-			if (str[1] < '0' || str[1] > '9') goto BOGUS;
-			if (str[2] != ':') goto BOGUS;
-			if (str[3] < '0' || str[3] > '9') goto BOGUS;
-			if (str[4] < '0' || str[4] > '9') goto BOGUS;
-			if (str[5] != ':') goto BOGUS;
-			if (str[6] < '0' || str[6] > '9') goto BOGUS;
-			if (str[7] < '0' || str[7] > '9') goto BOGUS;
-			MIN = (byte)((str[0] - '0') * 10 + (str[1] - '0'));
-			SEC = (byte)((str[3] - '0') * 10 + (str[4] - '0'));
-			FRAC = (byte)((str[6] - '0') * 10 + (str[7] - '0'));
-			Valid = true;
-			Negative = false;
-			return;
-		BOGUS:
-			MIN = SEC = FRAC = 0;
-			Valid = false;
+			Valid = str.Length == 8 && str[0] >= '0' && str[0] <= '9' && str[1] >= '0' && str[1] <= '9' &&
+				str[2] == ':' && str[3] >= '0' && str[3] <= '9' && str[4] >= '0' && str[4] <= '9' &&
+				str[5] == ':' && str[6] >= '0' && str[6] <= '9' && str[7] >= '0' && str[7] <= '9';

```

Comment:
If you move `str.Length == 8`, `str[2] == ':'`, and `str[5] == ':'` to the end of the corresponding lines, it'd allow to align the lines better, which helps to understand what's going on. Removal of goto should improve readability, so let's actually improve it.

Reply:
That's only because GitHub displays tabs as 8 x's wide though, the "s" starting line 113 is aligned with the "d" on line 112 in my IDE.

Reply:
This would look nicer IMO:
```
Valid =
	str[0] >= '0' && str[0] <= '9' && str[1] >= '0' && str[1] <= '9' && str.Length == 8 &&
	str[3] >= '0' && str[3] <= '9' && str[4] >= '0' && str[4] <= '9' && str[2] == ':' &&
	str[6] >= '0' && str[6] <= '9' && str[7] >= '0' && str[7] <= '9' str[5] == ':';
```

Reply:
Gonna call that out of scope :P

Reply:
What do you mean?

Reply:
This may be one of those situations where use of `goto` is preferable purely from a readability perspective...

Reply:
@vadosnaprimer I mean I couldn't be bothered. Changes made.